src/components/TokenLogo.vue

```diff
@@ -32,24 +23,59 @@ import type { WhitelistIdsBySymbol } from '../types/common';
   },
 })
 export default class TokenLogo extends Mixins(TranslationMixin) {
-  @getter.account.whitelist whitelist!: Whitelist;
-  @getter.account.whitelistIdsBySymbol whitelistIdsBySymbol!: WhitelistIdsBySymbol;
-  @mutation.router.navigate private navigate!: (options: Route) => void;
+  @getter.account.whitelist private whitelist!: Whitelist;
+  @getter.account.whitelistIdsBySymbol private whitelistIdsBySymbol!: WhitelistIdsBySymbol;
 
   @Prop({ type: String, default: '' }) readonly tokenSymbol!: string;
-  @Prop({ type: Object, default: ObjectInit }) readonly token!: AccountAsset | Asset;
+  @Prop({ type: Object, default: ObjectInit }) readonly token!: Nullable<AccountAsset | Asset>;
   @Prop({ type: String, default: LogoSize.MEDIUM, required: false }) readonly size!: LogoSize;
   @Prop({ default: false, type: Boolean }) readonly withClickableLogo!: boolean;
 
+  get isNft(): boolean {
+    return !!this.token && api.assets.isNft(this.token);
+  }
+
+  get assetAddress(): Nullable<string> {
+    return this.tokenSymbol ? this.whitelistIdsBySymbol[this.tokenSymbol] : (this.token || {}).address;
+  }
+
+  get whitelistedItem(): Nullable<WhitelistItem> {
+    if (!(this.token || this.tokenSymbol)) {
+      return null;
+    }
+    const address = this.assetAddress;
+    if (!address) {
+      return null;
+    }
+    return this.whitelist[address];
+  }
+
   get iconStyles(): object {
-    return getAssetIconStyles(this.token.address);
+    const address = this.assetAddress;
+    if (!address) {
+      return {};
+    }
```

Comment:
Could we remove this lines, since they are exists in `whitelistedItem`?
```
const address = this.assetAddress;
if (!address) {
  return {};
}
```